### PR TITLE
PR: Fix error when printing source code

### LINKED ISSUE #4824: Error when trying to print source code - AttributeError: 'QPrintDialog' object has no attribute 'addEnabledOption'
## Description

I want to print a source code file.  When I go to File->Print in Spyder I see the error message that I presume is linked to this in some manner.

**What steps will reproduce the problem?**

It doesn't seem reproducible in that I've now closed and re-opened the file and the print function works as expected.  To get this issue I opened the source file, and then I highlight items near the top of the file like "from utilities import *" and right-clicked on utilities to choose go to definition, as I wanted to also view those files.  I then went to File->Print and this error was generated.  I can't supply any of the files for commericial-in-confidence reasons.

**What is the expected output? What do you see instead?**
I expected to see a file dialog, instead I got the error below.

**Please provide any additional information below**

  File "C:\Users\fra256\AppData\Local\Continuum\Anaconda3\lib\site-packages\spyder\plugins\editor.py", line 1932, in print_file
    printDialog.addEnabledOption(QAbstractPrintDialog.PrintSelection)
AttributeError: 'QPrintDialog' object has no attribute 'addEnabledOption'

## Version and main components

* Spyder Version: 3.2.0 
* Python Version: 3.6.0
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.5.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.1.3 (OK)
pandas >=0.13.1  :  0.19.2 (OK)
numpy >=1.7      :  1.11.3 (OK)
sphinx >=0.6.6   :  1.5.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
jedi >=0.9.0     :  0.9.0 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
sympy >=0.7.3    :  1.0 (OK)
cython >=0.21    :  0.25.2 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
IPython >=4.0    :  5.1.0 (OK)
pylint >=0.25    :  1.6.4 (OK)

```

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #4824

### CODE REVIEW

File: spyder/plugins/editor.py [hunk 1]
```diff
@@ -1929,7 +1929,7 @@ def print_file(self):
                           header_font=self.get_plugin_font('printer_header'))
         printDialog = QPrintDialog(printer, editor)
         if editor.has_selected_text():
-            printDialog.addEnabledOption(QAbstractPrintDialog.PrintSelection)
+            printDialog.setOption(QAbstractPrintDialog.PrintSelection, True)
```
Comment (goanpeca) [inline spyder/plugins/editor.py]: @dalthviz is this a change on Qt5 ? or ?, should we had some try/except based on version number?

Comment (dalthviz) [inline spyder/plugins/editor.py]: From what I have read in the docs, `addEnabledOption` is and obsolet member of the class `QAbstractPrintDialog` (from QPrintDialog inherits), here the [link to the docs of Qt 4.8](https://doc.qt.io/qt-4.8/qabstractprintdialog-obsolete.html#addEnabledOption)

Comment (goanpeca) [inline spyder/plugins/editor.py]: @dalthviz can we check that it will work on Qt4 and Qt5, and try to add a small test for it?

Comment (ccordoba12) [inline spyder/plugins/editor.py]: Is this method in Qt5 as well?

Comment (dalthviz) [inline spyder/plugins/editor.py]: yep, [link to the docs](http://doc.qt.io/qt-5/qprintdialog.html#setOption)

**Comment by ccordoba12** [comment]
@dalthviz, the test you introduced is hanging on Travis. Please take a look at that.

**Pull request merged**